src/entities/position/ui/table.tsx

```diff
@@ -0,0 +1,271 @@
+    void fetchNextPage();
+  });
+
+  const [sortBy, setSortBy] = useState<{
+    key: string;
+    direction: 'desc' | 'asc';
+  }>({
+    key: 'effectivePrice',
+    direction: 'desc',
+  });
+
+  const sortedPositions = useMemo<DisplayPosition[]>(() => {
+    return orderBy([...displayPositions], `sortValues.${sortBy.key}`, sortBy.direction);
+  }, [displayPositions, sortBy]);
+
+  const SortableTableHeader = useCallback(
+    ({ sortKey, children }: { sortKey: string; children: ReactNode }) => {
+      return (
+        <TableCell heading>
+          <button
+            className={cn(
+              'flex bg-none border-none',
+              sortBy.key === sortKey ? 'text-text-primary' : 'text-text-secondary',
+            )}
+            onClick={() => {
+              setSortBy({
+                key: sortKey,
+                direction: sortBy.key === sortKey && sortBy.direction === 'desc' ? 'asc' : 'desc',
+              });
+            }}
+          >
+            <Text tableHeadingSmall whitespace='nowrap'>
+              {children}
+            </Text>
+            {sortKey === sortBy.key && (
+              <>
+                {sortBy.direction === 'asc' ? (
+                  <ChevronUp className='w-4 h-4' />
+                ) : (
+                  <ChevronDown className='w-4 h-4' />
+                )}
+              </>
+            )}
+          </button>
+        </TableCell>
+      );
+    },
+    [sortBy, setSortBy],
+  );
+
+  const loadingArr = new Array(5).fill({
+    position: {},
+    orders: [
+      {
+        baseAsset: {
+          asset: {},
+        },
+        quoteAsset: {
+          asset: {},
+        },
+      },
+    ],
+  }) as DisplayPosition[];
+
+  if (!connected) {
+    return <NotConnectedNotice />;
+  }
+
+  if (error) {
+    return <ErrorNotice />;
+  }
+
+  if (!isLoading && !sortedPositions.length) {
+    return <NoPositions />;
+  }
+
+  return (
+    <div className='grid grid-cols-[80px_1fr_1fr_80px_1fr_1fr_1fr_1fr] overflow-y-auto overflow-x-auto'>
+      <Density slim>
+        <div className='grid grid-cols-subgrid col-span-8'>
+          <SortableTableHeader sortKey='type'>Type</SortableTableHeader>
+          <SortableTableHeader sortKey='tradeAmount'>Trade Amount</SortableTableHeader>
+          <SortableTableHeader sortKey='effectivePrice'>Effective Price</SortableTableHeader>
+          <SortableTableHeader sortKey='feeTier'>Fee Tier</SortableTableHeader>
+          <SortableTableHeader sortKey='basePrice'>Base Price</SortableTableHeader>
+          <TableCell heading>Current Value</TableCell>
+          <SortableTableHeader sortKey='positionId'>Position ID</SortableTableHeader>
+          <TableCell heading>
+            <HeaderActionButton displayPositions={sortedPositions} />
+          </TableCell>
+        </div>
+
+        {(isLoading ? loadingArr : sortedPositions).map((position, index) => (
+          <Fragment key={`${position.idString}${index}`}>
```

Comment:
You don't need to do this btw, keys only have to be unique for each loop. So `index` is enough here incase we have multiple positions with the same `idString`.